File: 큐와 덱/1966/1966.cpp
#include <algorithm>
#include <iostream>
#include <queue>
using namespace std;

int main()
{
   queue<vector<int>> q;
   priority_queue<int> pq;
   int t, n, m, temp;
   cin>>t;
   for (int i = 0; i < t; ++i)
   {
      cin>>n>>m;
      for (int j = 0; j < n; ++j)
      {
         cin>>temp;
         q.push({temp, j});
         pq.push(temp);
      }
      int j = 1;
      while (!q.empty())
      {
         if (q.front()[0] == pq.top())
         {
            if (m == q.front()[1])
            {
               cout<<j<<"\n";
               break;
            }
            ++j;
            q.pop();
            pq.pop();
         }
         else
         {
            q.push(q.front());
            q.pop();
         }
      }
      while (!q.empty())
         q.pop();
      while (!pq.empty())
         pq.pop();
   }

   return 0;
}

File: 큐와 덱/1966/1966.py
from queue import Queue
from queue import PriorityQueue

q = Queue()
pq = PriorityQueue()
t = int(input())
for i in range(t):
   n, m = map(int, input().split())
   temp = list(map(int, input().split()))
   for j in range(len(temp)):
      q.put([temp[j], j])
      pq.put(-temp[j])
   j = 1
   while not q.empty():
      if q.queue[0][0] == -pq.queue[0]:
         if m == q.queue[0][1]:
            print(j)
            break
         j += 1
         q.get()
         pq.get()
      else:
         q.queue.rotate(-1)
   q.queue.clear()
   pq.queue.clear()